src/instance/mod.rs

```diff
@@ -128,3 +166,66 @@ pub trait ToObjectInstance {
 pub trait FromObjectInstance {
     fn from_instance(i: &ObjectInstance) -> Self;
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{parser::Construct, Context, JkInt};
+
+    fn setup() -> Context {
+        let mut ctx = Context::new();
+
+        let inst = Construct::instruction("type Point(x: int, y: int); ")
+            .unwrap()
+            .1;
+        inst.execute(&mut ctx);
+
+        let inst = Construct::instruction("type Vec2(f: Point, s: Point); ")
+            .unwrap()
+            .1;
+        inst.execute(&mut ctx);
+
+        let inst = Construct::instruction("p = Point { x = 1, y = 2}")
+            .unwrap()
+            .1;
+        inst.execute(&mut ctx);
+
+        let inst = Construct::instruction("v = Vec2 { f = p, s = p}")
+            .unwrap()
+            .1;
+        inst.execute(&mut ctx);
+
+        ctx
```

Comment:
Indeed a macro would be kind of neat, similar to asm! have a jinko!
A loop that executes an array of instructions is probably easier though

Reply:
Yeah, there's an issue for that (#220) and I really want to work on it but it's not a priority. I think it would be quite easy to do too, and super helpful for testing as well as for using jinko as a crate